src/ActiveLogin.Authentication.BankId.Api/Models/SignRequest.cs

```diff
@@ -0,0 +1,139 @@
+        /// Data not displayed to the user. String. The value must be base 64-encoded. 1-200 000charactersafter base 64-encoding.
+        /// </param>
+        /// <param name="requirement">Requirements on how the auth or sign order must be performed.</param>
+        public SignRequest(string endUserIp, string personalIdentityNumber, string userVisibleData, string userNonVisibleData, Requirement requirement)
+        {
+            EndUserIp = endUserIp;
+            PersonalIdentityNumber = personalIdentityNumber;
+            Requirement = requirement;
+            UserVisibleData = EnsureBase64EncodedString(userVisibleData);
+            UserNonVisibleData = EnsureBase64EncodedString(userNonVisibleData);
+        }
+
+        /// <summary>
+        /// The user IP address as seen by RP. IPv4 and IPv6 is allowed.
+        /// Note the importance of using the correct IP address.It must be the IP address representing the user agent (the end user device) as seen by the RP.
+        /// If there is a proxy for inbound traffic, special considerations may need to be taken to get the correct address.
+        ///
+        /// In some use cases the IP address is not available, for instance for voice based services.
+        /// In this case, the internal representation of those systems IP address is ok to use.
+        /// </summary>
+        [DataMember(Name = "endUserIp")]
+        public string EndUserIp { get; private set; }
+
+        /// <summary>
+        /// The personal number of the user. 12 digits, century must be included (YYYYMMDDSSSC).
+        /// If the personal number is excluded, the client must be started with the AutoStartToken returned in the response.
+        /// </summary>
+        [DataMember(Name = "personalNumber", EmitDefaultValue = false)]
+        public string PersonalIdentityNumber { get; private set; }
+
+        /// <summary>
+        /// Requirements on how the auth or sign order must be performed.
+        /// </summary>
+        [DataMember(Name = "requirement", EmitDefaultValue = false)]
+        public Requirement Requirement { get; private set; }
+
+        /// <summary>
+        /// The text to be displayed and signed. String. The text can be formatted using CR, LF and CRLF for new lines.
+        /// The text must be encoded as UTF-8 and then base 64 encoded. 1--40 000 characters after base 64 encoding.
+        /// </summary>
+        [DataMember(Name = "userVisibleData")]
+        public string UserVisibleData { get; private set; }
+
+        /// <summary>
+        /// Data not displayed to the user. String. The value must be base 64-encoded. 1-200 000 characters after base 64-encoding.
+        /// </summary>
+        [DataMember(Name = "userNonVisibleData", EmitDefaultValue = false)]
+        public string UserNonVisibleData { get; private set; }
+
+        private static string EnsureBase64EncodedString(string value)
+        {
+            if (value == null) return null;
```

Comment:
This isn't documented, but we usually do multi line if statements for this, so:
```suggestion
            if (value == null) {
                return null;
            }
```